lib/src/code_generator/objc_block.dart

```diff
@@ -115,12 +115,20 @@ class $name extends _ObjCBlockBase {
   static $voidPtr? _cFuncTrampoline;
 
   /// Creates a block from a Dart function.
-  $name.fromFunction(${w.className} lib, ${funcType.getDartType(w)} fn) :
-      this._(lib.${builtInFunctions.newBlock.name}(
+  static $name fromFunction(${w.className} lib, ${funcType.getDartType(w)} fn) {
+    final oldBlock = _funcToBlock[fn];
+    if (oldBlock != null) {
+      return $name._(oldBlock, lib, retain: true);
+    }
+    final newBlock = lib.${builtInFunctions.newBlock.name}(
           _dartFuncTrampoline ??= ${w.ffiLibraryPrefix}.Pointer.fromFunction<
               ${trampFuncType.getCType(w)}>($closureTrampoline
-                  $exceptionalReturn).cast(), $registerClosure(fn)), lib);
+                  $exceptionalReturn).cast(), $registerClosure(fn));
+    _funcToBlock[fn] = newBlock;
+    return $name._(newBlock, lib);
```

Comment:
We don't increase refcount for `newBlock` here. 

What happens if the underlying obj-c block is de-rereferenced and freed while still being in the expando map? If one creates another block for same closure, we have a use-after-free, don't we?

I'll comment more on the github issue

Reply:
I guess the assumption is that the dart wrapper will keep a refcount as long as it's alive?

Reply:
Yeah, the Dart wrapper always holds one reference. In all the other cases where we construct the wrapper function, we're using a block pointer from Block_copy, which already has an incremented ref count.